### PR TITLE
po: Build translations with webpack

### PR DESCRIPTION
**Pull request by martinpitt**
Convert the `po2json` script into a webpack plugin, and integrate the
(now trivial) po.empty.js template.

This is another step for building the entire dist/ directory with
`npm run build` (i.e. a tool that web developers are familiar with),
and not having a split webpack+make toolchain any more.

Based on starter-kit commit 7f6ef51c12a75. cockpit-po-plugin.js was
extended to accept a `subdir` option for building a particular
pkg/<page>/ only, a `msggrep_options` option for additional `msggrep`
command line arguments, and a `wrapper` option for using a custom call
to poke the PO data into the page (which we need for the login page, as
that does not have cockpit.js). These are backwards compatible with thhe
starter-kit version, so in the future we can drop the latter copies and
replace them with a tagged git checkout.

Change how ws inserts translations into the login page: Previously, the
Makefile generated static/login.po.LL.html files *in addition* to the
usual po.LL.js files, which duplicated the translations again. Drop all
these, and make ws insert the po.*.js file directly instead. Add a new
marker so that this happens inside a `<script>` tag.

Note: It would be way more elegant to just make login.html include
`<script src="po.js"></script>`, but unfortunately that does not work
with ws (it does not pick the correct per-language po.LL.js file in the
login page code paths).

----

I compared translations against master with `for f in dist/*/po.de.js; do diff -u ../cockpit/$f $f; done|view -` and the diff looks reasonable. In fact this *adds* a few forgotten translations from manifest descriptions and keywords.

 - [x] Builds on top of PR #15119
 - [x] Builds on top of PR #15142 
 - [x] Fix login page translations and broken layout
 - [x] Adjust commit message, we also need to build manifests with webpack
 - [x] Fix handling po.js.gz (ws can't transparently serve static .gz files)
 - [x] Fix [check-pages regressions](https://logs.cockpit-project.org/logs/pull-15120-20210114-104422-ee2afe9e-fedora-33/log.html): PR #15149

### CODE REVIEW

**Comment by martinpitt** [comment]
Yowch, we have so many effing special treatments of the login page.. I'd like to drop all of them, but I didn't manage to get it to work properly with the way every other page includes translations. So for now I just adapted the special case, and at least made it quite a bit smaller.

File: webpack.config.js [hunk 1]
```diff
@@ -248,14 +248,17 @@ const info = {
...
+
 const webpack = require("webpack");
+
 const copy = require("copy-webpack-plugin");
-const childProcess = require('child_process');
 const html = require('html-webpack-plugin');
 const miniCssExtractPlugin = require('mini-css-extract-plugin');
-const path = require("path");
-const fs = require("fs");
 const OptimizeCSSAssetsPlugin = require('optimize-css-assets-webpack-plugin');
```
Comment (martinpitt) [inline webpack.config.js:260]: Note: This is mostly noise. I wanted to sort the plugins into a separate block, and keep the "standard node library" modules at the top.

**Comment by martinpitt** [comment]
I fixed translations of login page with installed packages locally, but there are still [some test failures](https://logs.cockpit-project.org/logs/pull-15120-20210114-104422-ee2afe9e-fedora-33/log.html).

**Comment by martinpitt** [comment]
@skobyda: Do you know what [this Machines snapshot failure](https://logs.cockpit-project.org/logs/pull-15120-20210114-181354-1b91bfad-fedora-33/log.html#117-2) means? I doubt it is translation related.

**Comment by martinpitt** [comment]
Setting needswork, I need to look at TestMachines.testSnapshots.

**Comment by skobyda** [comment]
Hmmm, never seen that failure, but should be easy to fix, let me look into that (too bad that the screenshot of test failures do not cover the whole page. The problem would be better visible if it was scrolled a bit down :D )

**Comment by skobyda** [comment]
I sent a fix https://github.com/cockpit-project/cockpit/pull/15149 .

> I doubt it is translation related. it happens on [debian-testing](https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/pull-15120-20210114-190302-1b91bfad-debian-testing/log.html) as well.

Actually, yhe problem might as well related to localization. The MomentJS is used to format the date string (which localises it). And with this PR, on some distros the format changed from "Today 11:25 PM" to "Today at 11:25 PM", so perhaps it could be related to that? But nevermind, I changed the test so it works with both formats.

**Comment by martinpitt** [comment]
FTR, I tested this with the translation update in PR #15144 and it works fine. I tested the login page interactively, and it is fully translated.

**Comment by martinpitt** [comment]
Oh noes, conflict with Lis'es NO_NPM PR that just landed. Too bad for the perfectly green test result here.. Rebased.

**Comment by martinpitt** [comment]
I added another commit that updates the *.po files (`make update-po`), so that the file paths are correct everywhere. This affected master as well, but let's do that while we are at it. Thanks @marusak for spotting this!

**Review by marusak — APPROVED** [review]
Nice, thanks!
It fixes both broken login page translations and also not all labels being translated, great work!

**Comment by marusak** [comment]
Failing tests are all well known flakes, most of them are on our Friday Flakes list :)

**Pull request merged**